app/front-end/src/features/editor/components/toolbarView/toolbarGroupParams/applyGroupParams.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  StyledGroupParamsInputLabel,
+  StyledGroupParamsListSubheader,
+  StyledGroupParamsMenuItem,
+  StyledGroupParamsMenuItemTypography,
+  StyledGroupParamsMenuItemTypographyBold,
+  StyledGroupParamsSelect,
+  StyledGroupParamsTypography,
+} from '@/features/editor/components/toolbarView/toolbarGroupParams';
+import { useToolbarContext, useWorkspaceContext } from '@/features/editor/hooks';
+import { FileModel, FileTypes } from '@/features/editor/types';
+import { getWorkspaceArray } from '@/features/editor/utils';
+import { useStatusContext } from '@/hooks';
+import { Box, Checkbox, FormControl, FormControlLabel, SelectChangeEvent, useTheme } from '@mui/material';
+import { useEffect, useState } from 'react';
+
+export interface ApplyGroupParamsProps {}
+
+export const ApplyGroupParams: React.FC<ApplyGroupParamsProps> = () => {
+  const { blocked } = useStatusContext();
+  const Theme = useTheme();
+  const { fileTree, fileTreeArray } = useWorkspaceContext();
+  const { saveTo, saveToStateUpdate, applyTo, applyToStateUpdate, applyError, applyErrorStateUpdate } =
+    useToolbarContext();
+
+  //
+  // Apply state
+  //
+  const [applyToValue, setApplyToValue] = useState<string>(applyTo);
+
+  const handleApplyToChange = (event: SelectChangeEvent<any>) => {
+    setApplyToValue(event.target.value);
+    applyToStateUpdate(event.target.value);
+    applyErrorStateUpdate('');
+  };
+
+  //
+  // Save To state
+  //
+  const [fileArray, setFileArray] = useState<FileModel[]>(fileTreeArray);
+  const [saveToValue, setSaveToValue] = useState<string>(saveTo);
+  const [overrideValue, setOverrideValue] = useState<boolean>(false);
+
+  const handleSaveToChange = (event: SelectChangeEvent<any>) => {
+    setSaveToValue(event.target.value);
+    saveToStateUpdate(event.target.value, false);
+    setOverrideValue(false);
+  };
+
+  const handleOverrideChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setOverrideValue(event.target.checked);
+    saveToStateUpdate(saveToValue, event.target.checked);
+  };
+
+  //
+  // Effects
+  //
+  useEffect(() => {
+    setFileArray(getWorkspaceArray(fileTree));
+  }, [fileTree]);
+
+  return (
+    <Box sx={{ display: 'grid', gridTemplateColumns: '50% 50%', p: '1rem' }}>
+      <Box
+        sx={{
+          height: '100%',
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'start',
+          gap: '1rem',
+        }}
+      >
+        <FormControl sx={{ width: '90%' }}>
+          <StyledGroupParamsInputLabel
+            sx={{
+              color: blocked
+                ? Theme.palette.action.disabled
+                : applyError
+                  ? Theme.palette.error.main
+                  : Theme.palette.text.primary,
+            }}
+          >
```

Comment:
Since it's being reused elsewhere in the code, maybe it should've been made as a seperate component with parameters? Passing 'error' parameter would be enough since it could take 'blocked' from context.

Reply:
Fixed with [MDE/PKFE-31 params additional fixes and changes](https://github.com/Strexas/kath/pull/57/commits/6c8fad426f15d0b9d6eec084d13a931512fcfb2d)